src/NumericalAlgorithms/Spectral/Spectral.cpp

```diff
@@ -149,6 +152,16 @@ struct DifferentiationMatrixGenerator {
   }
 };
 
+template <Basis BasisType, Quadrature QuadratureType>
+struct IntegrationMatrixGenerator {
+  Matrix operator()(const size_t num_points) const noexcept {
+    const Mesh<1> mesh(num_points, BasisType, QuadratureType);
+    return Spectral::spectral_to_grid_points_matrix(mesh) *
```

Comment:
call the `Spectral::spectral_to_grid_points_matrix<BasisType,QuadratureType>` template directly. Then you can eliminate the runtime mesh.